contracts/connection-router/src/state.rs

```diff
@@ -81,26 +82,61 @@ impl<'a> IndexList<ChainEndpoint> for ChainEndpointIndexes<'a> {
     }
 }
 
-// Message represents a message for which the fields have been successfully validated.
-// This should never be supplied by the user.
 #[cw_serde]
 pub struct NewMessage {
-    pub id_on_chain: MessageID, // globally unique
-    pub destination_address: Address,
+    pub id_on_chain: MessageID, // unique per chain
```

Comment:
Are we changing this to be only unique per chain now?

Reply:
I strongly advocate for it for the following reasons:
- with the global id, the original producer is tied to the inner state checks of our contracts, because it needs to know to create this global id, instead of just putting the unique id it gets from the chain
- because source_chain and id can possibly clash, you can either guarantee safety of the type by making id private and only providing setters/getters (making it awkward to create) or making it easy to create messages but losing safety
- the global id is easily derived. As a bonus, the id is now responsible for it's own representation and acts as a composite key, so we can actually iterate over all messages of a particular chain if we need to